packages/ckeditor5-dev-docs/lib/index.js

```diff
@@ -34,35 +38,59 @@ function build( config ) {
 		process.env.JSDOC_VALIDATE_ONLY = true;
 	}
 
-	const jsDocConfig = {
-		opts: {
-			encoding: 'utf8',
-			recurse: true,
-			access: 'all',
-			template: 'templates/silent'
-		},
-		plugins: [
-			require.resolve( 'jsdoc/plugins/markdown' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/export-fixer/export-fixer' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/custom-tags/error' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/custom-tags/observable' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/observable-event-provider' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/relation-fixer' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/longname-fixer/longname-fixer' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/event-extender/event-extender' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/validator/validator' ),
-			require.resolve( '@ckeditor/jsdoc-plugins/lib/utils/doclet-logger' )
-		]
-	};
+	const files = [];
 
 	return new Promise( ( resolve, reject ) => {
-		gulp.src( sourceFiles, { read: false } )
-			.pipe( jsdoc( jsDocConfig, result => {
-				if ( result instanceof Error ) {
-					return reject( result.message );
-				}
+		vfs.src( sourceFiles )
+			.pipe( map( ( file, callback ) => {
+				files.push( file.path );
+				callback( null, file );
+			} ).on( 'end', () => {
+				const jsDocConfig = {
+					plugins: [
+						require.resolve( 'jsdoc/plugins/markdown' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/export-fixer/export-fixer' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/custom-tags/error' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/custom-tags/observable' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/observable-event-provider' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/relation-fixer' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/longname-fixer/longname-fixer' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/event-extender/event-extender' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/validator/validator' ),
+						require.resolve( '@ckeditor/jsdoc-plugins/lib/utils/doclet-logger' )
+					],
+					source: {
+						include: files
+					},
+					opts: {
+						encoding: 'utf8',
+						recurse: true,
+						access: 'all',
+						template: 'templates/silent'
+					}
+				};
+
+				const tmpConfig = tmp.fileSync();
+
+				fs.writeFile( tmpConfig.name, JSON.stringify( jsDocConfig ), 'utf8', error => {
+					if ( error ) {
+						return reject( error );
+					}
+
+					const cmd = require.resolve( 'jsdoc/jsdoc.js' );
+
+					console.log( 'JSDoc started...' );
+
+					try {
+						tools.shExec( `${ cmd } -c ${ tmpConfig.name }`, { verbosity: 'info' } );
+					} catch ( error ) {
+						return reject( 'Error during JSDoc generation.' );
```

Comment:
Would be nice to attach message from `error`. It should help understand what went wrong.